src/PrestaShopBundle/Translation/DTO/Translations.php

```diff
@@ -0,0 +1,171 @@
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Translation\DTO;
+
+class Translations
+{
+    public const METADATA_KEY_NAME = '__metadata';
+
+    /**
+     * @var DomainTranslation[]
+     */
+    private $domainTranslations;
+
+    public function __construct()
+    {
+        $this->domainTranslations = [];
+    }
+
+    /**
+     * @param DomainTranslation $domainTranslations
+     *
+     * @return Translations
+     */
+    public function addDomainTranslation(DomainTranslation $domainTranslations): self
+    {
+        if (!array_key_exists($domainTranslations->getDomainName(), $this->domainTranslations)) {
+            $this->domainTranslations[$domainTranslations->getDomainName()] = $domainTranslations;
+        }
+
+        return $this;
+    }
+
+    /** Returns a single DomainTranslation DTO.
+     *
+     * @param string $domainName
+     *
+     * @return DomainTranslation|null
+     */
+    public function getDomainTranslation(string $domainName): ?DomainTranslation
+    {
+        if (array_key_exists($domainName, $this->domainTranslations)) {
+            return $this->domainTranslations[$domainName];
+        }
+
+        return null;
+    }
+
+    /**
+     * @return DomainTranslation[]
+     */
+    public function getDomainTranslations(): array
+    {
+        return $this->domainTranslations;
+    }
+
+    public function getTranslationsCount(): int
+    {
+        $countTranslations = 0;
+        foreach ($this->domainTranslations as $domainTranslation) {
+            $countTranslations += $domainTranslation->getTranslationsCount();
+        }
+
+        return $countTranslations;
+    }
+
+    public function getMissingTranslationsCount(): int
+    {
+        $missingTranslations = 0;
+        foreach ($this->domainTranslations as $domainTranslation) {
+            $missingTranslations += $domainTranslation->getMissingTranslationsCount();
+        }
+
+        return $missingTranslations;
+    }
+
+    /**
+     * @return array
+     */
+    public function toArray(bool $withMetadata = true): array
+    {
+        $data = [];
+        foreach ($this->domainTranslations as $domainTranslation) {
+            $data[$domainTranslation->getDomainName()] = $domainTranslation->toArray($withMetadata);
+        }
+
+        if ($withMetadata) {
+            $data[self::METADATA_KEY_NAME] = [
+                'count' => count($this->domainTranslations),
+                'missing_translations' => $this->getMissingTranslationsCount(),
+            ];
+        }
+
+        ksort($data);
+
+        return $data;
+    }
+
+    public function getTree(): array
```

Comment:
```suggestion
    public function buildAsATree(): array
```
Would it be over-thinking to do that? 🤔